sway-core/src/semantic_analysis/node_dependencies.rs

```diff
@@ -907,6 +908,7 @@ fn type_info_name(type_info: &TypeInfo) -> String {
         TypeInfo::Slice(..) => "__slice",
         TypeInfo::Alias { .. } => "alias",
         TypeInfo::TraitType { .. } => "trait type",
+        TypeInfo::Ref(..) => "ref",
```

Comment:
Given that this is used as key in trees, shouldn't it have the `referred type`? This method return `String`, so I think you are safe to append it here.

Reply:
The returned string needs to be different from the one used to display `TypeInfo` which is "&". So "ref" should be fine here. Still, to be on the safe side, I've changed it to "reference type" :-)